main.go

```diff
@@ -490,8 +495,9 @@ func main() {
 		})
 		mux.HandleFunc("/-/quit", func(w http.ResponseWriter, r *http.Request) {
 			if r.Method == http.MethodPut || r.Method == http.MethodPost {
+				fmt.Fprintf(w, "Requesting termination... Goodbye!")
 				level.Info(logger).Log("msg", "Received lifecycle api quit, exiting")
-				os.Exit(0)
+				signals <- os.Kill
```

Comment:
I'm curious how this works, since we do not `.Notify` on this signal in particular (IIUC it's not normally one that can be caught)? To me it would be clearer to send `os.Interrupt` here, would that have unintended side effects?

Reply:
You are correct in that we only notify that channel on certain signals, but in this case it doesn't matter what signal we send to the channel, as anything sent to that channel will cause it to quit. I can add a quit channel if that would be a little clearer

Reply:
ah, that makes sense, thank you!